Source/VMware.vSphereDSC/DSCResources/PowerCLISettings.ps1

```diff
@@ -0,0 +1,235 @@
+        Invoke-Expression -Command $constructedCommand
+    }
+
+    [bool] Test() {
+        $this.ImportRequiredModules()
+        $powerCLICurrentConfiguration = Get-PowerCLIConfiguration -Scope $this.Scope
+        $powerCLIDesiredConfiguration = $this.GetPowerCLIConfigurationProperties()
+
+        return $this.Equals($powerCLICurrentConfiguration, $powerCLIDesiredConfiguration)
+    }
+
+    [PowerCLISettings] Get() {
+        $this.ImportRequiredModules()
+        $result = [PowerCLISettings]::new()
+        $powerCLICurrentConfiguration = Get-PowerCLIConfiguration -Scope $this.Scope
+
+        $this.PopulateResult($powerCLICurrentConfiguration, $result)
+
+        return $result
+    }
+
+    <#
+    .DESCRIPTION
+
+    Imports the needed VMware Modules.
+    #>
+    [void] ImportRequiredModules() {
+        $savedVerbosePreference = $global:VerbosePreference
+        $global:VerbosePreference = 'SilentlyContinue'
+
+        Import-Module -Name VMware.VimAutomation.Core
+
+        $global:VerbosePreference = $savedVerbosePreference
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns all passed PowerCLI configuration properties as a hashtable.
+    #>
+    [hashtable] GetPowerCLIConfigurationProperties() {
+        $powerCLIConfigurationProperties = @{}
+
+        # Adds the Default Scope to the hashtable.
+        $powerCLIConfigurationProperties.Add("Scope", $this.Scope)
+
+        if ($this.CEIPDataTransferProxyPolicy -ne [ProxyPolicy]::Unset -and $this.ParticipateInCeip -eq $true) {
+            $powerCLIConfigurationProperties.Add("CEIPDataTransferProxyPolicy", $this.CEIPDataTransferProxyPolicy)
+        }
+
+        if ($this.DefaultVIServerMode -ne [DefaultVIServerMode]::Unset) {
+            $powerCLIConfigurationProperties.Add("DefaultVIServerMode", $this.DefaultVIServerMode)
+        }
+
+        if ($null -ne $this.DisplayDeprecationWarnings) {
+            $powerCLIConfigurationProperties.Add("DisplayDeprecationWarnings", $this.DisplayDeprecationWarnings)
+        }
+
+        if ($this.InvalidCertificateAction -ne [BadCertificateAction]::Unset) {
+            $powerCLIConfigurationProperties.Add("InvalidCertificateAction", $this.InvalidCertificateAction)
+        }
+
+        if ($null -ne $this.ParticipateInCeip) {
+            $powerCLIConfigurationProperties.Add("ParticipateInCeip", $this.ParticipateInCeip)
+        }
+
+        if ($this.ProxyPolicy -ne [ProxyPolicy]::Unset) {
+            $powerCLIConfigurationProperties.Add("ProxyPolicy", $this.ProxyPolicy)
+        }
+
+        if ($null -ne $this.WebOperationTimeoutSeconds) {
+            $powerCLIConfigurationProperties.Add("WebOperationTimeoutSeconds", $this.WebOperationTimeoutSeconds)
+        }
+
+        return $powerCLIConfigurationProperties
+    }
+
+    <#
+    .DESCRIPTION
+
+    Constructs the Set-PowerCLIConfiguration cmdlet with the passed properties.
```

Comment:
Add detailed comment here and where this function is used why it is needed to be used this way instead of splatting.